### PR TITLE
Feat/0004/frontend connect

### PR DESCRIPTION
**Pull request by ayushmi173**
- Set up frontend code with next js with typescript
- store data in global container using redux-thunk
- communicate login and signup form with api
- use styled component for styling

### CODE REVIEW

File: auth-connect-frontend/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  font-weight: 600;
+`;
+const Home = () => {
+  return (
+    <>
+      <Head>
+        <title>Connect Authentication SDK</title>
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <Layout title={"Home"} />
```
Comment (nautidpk) [inline auth-connect-frontend/pages/index.tsx:20]: if string as a prop directly pass this as `title="Home"`

File: auth-connect-frontend/pages/profile.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { UserProfile } from "../src/components/user-profile";
+
+const Profile = () => {
+  return (
+    <>
+      <Head>
+        <title>Connect Authentication SDK</title>
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <Layout title={"profile"} />
```
Comment (nautidpk) [inline auth-connect-frontend/pages/profile.tsx:13]: if string as a prop directly pass this as title="Home"

File: auth-connect-frontend/pages/signin.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import Layout from "../src/components/shared/layout";
+import { LogIn } from "../src/components/login";
+
+const SignIn = () => {
+  return (
+    <>
+      <Layout title={"Sign In"} />
+      <LogIn />
```
Comment (nautidpk) [inline auth-connect-frontend/pages/signin.tsx:8]: what is difference b/w signin and login?

File: auth-connect-frontend/pages/signup.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Register } from "../src/components/register";
+import Layout from "../src/components/shared/layout";
+
+const SignUp = () => {
+  return (
+    <>
+      <Layout title={"Sign Up"} />
+      <Register />
```
Comment (nautidpk) [inline auth-connect-frontend/pages/signup.tsx:9]: what is difference b/w signup and register??

File: auth-connect-frontend/src/components/login.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  min-height: 80vh;
+`;
+
+type Props = {};
+
+export const LogIn: React.FC<Props> = (props: Props) => {
+  const [username, setUsername] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const dispatch = useDispatch();
+  const ErrorMessage = useSelector((state: InitialEntityState) => state.error);
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/login.tsx:24]: camelCase variable naming

File: auth-connect-frontend/src/components/login.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+          {passwordInput}
+        </Form>
+        {ErrorMessage ? (
+          <Message summary={"There is something error"} />
+        ) : (
+          <>
+            {accessToken ? (
+              <Message summary={"Successfully Logged In.. Look Your Profile"} />
+            ) : (
+              <></>
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/login.tsx:84]: what is this doing??

File: auth-connect-frontend/src/components/register.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      type={"email"}
+      placeholder={"Enter Email"}
+      value={email}
+      disabled={false}
+      key={3}
+      onChange={(value) => setEmail(value)}
+    />
+  );
+
+  const clearFields = () => {
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/register.tsx:70]: why store jsx as variable ??`

File: auth-connect-frontend/src/components/register.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>): void {
+    event.preventDefault();
+    dispatch(userSignUp(username, password, email));
+    if (!ErrorMessage) {
+      clearFields();
+    }
+  }
+
+  function handleReset(event: React.MouseEvent<HTMLButtonElement>): void {
+    clearFields();
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/register.tsx:85]: why we need to reset state ?
cant we just move to next page after signing in ?

File: auth-connect-frontend/src/components/login.tsx [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+type Props = {};
+
+export const LogIn: React.FC<Props> = (props: Props) => {
+  const [username, setUsername] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const dispatch = useDispatch();
+  const ErrorMessage = useSelector((state: InitialEntityState) => state.error);
+  const accessToken = useSelector(
+    (state: InitialEntityState) => state.entities.accessToken
+  );
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/login.tsx:27]: why store accessToken in state?

File: auth-connect-frontend/src/components/register.tsx [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+type Props = {};
+
+export const Register: React.FC<Props> = (props: Props) => {
+  const [username, setUsername] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const [email, setEmail] = useState<string>("");
+
+  const dispatch = useDispatch();
+
+  const ErrorMessage = useSelector((state: InitialEntityState) => state.error);
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/register.tsx:28]: camelcase

File: auth-connect-frontend/src/components/shared/button.tsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import styled from "styled-components";
+import { COLORS } from "../../colors/constants";
+
+type Props = {
+  title: React.ReactNode;
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => any;
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/shared/button.tsx:7]: onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;

File: auth-connect-frontend/src/components/shared/button.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+  width: 110px;
+  height: 30px;
+  color: ${COLORS.WHITE};
+  background-color: ${COLORS.BLACK};
+  border-radius: 3px;
+`;
+export const Button: React.FC<Props> = (props: Props) => {
+  return (
+    <>
+      <ButtonWrapper>{props.title}</ButtonWrapper>
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/shared/button.tsx:22]: whre  onClick not added here??

File: auth-connect-frontend/src/components/shared/layout.tsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+  border: 1px solid ${COLORS.GREY};
+`;
+
+const NavWrapper = styled.nav`
+  display: flex;
+  justify-content: space-around;
+  flex-direction: row;
+`;
+
+const HeadWrapper = styled(Head)``;
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/shared/layout.tsx:23]: can we not use Head if no styling in it??

File: auth-connect-frontend/src/components/user-profile.tsx [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+  font-size: 30px;
+  color: ${COLORS.ERROR};
+`;
+
+export const UserProfile: React.FC<Props> = (props: Props) => {
+  const dispatch = useDispatch();
+  const accessToken = useSelector(
+    (state: InitialEntityState) => state.entities.accessToken
+  );
+  const userCredential = useSelector((state: InitialEntityState) => state.user);
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/user-profile.tsx:30]: userCredential = > user

File: auth-connect-frontend/src/components/user-profile.tsx [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+  );
+
+  const passwordInput = (
+    <TextInput
+      required={true}
+      label={"Password"}
+      type={"text"}
+      value={userCredential.password}
+      disabled={true}
+      key={2}
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/user-profile.tsx:61]: you dont have to store jsx as variable here

File: auth-connect-frontend/src/components/user-profile.tsx [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+  min-height: 80vh;
+`;
+
+const MessageWrapper = styled.span`
+  font-weight: 700;
+  font-size: 30px;
+  color: ${COLORS.ERROR};
+`;
+
+export const UserProfile: React.FC<Props> = (props: Props) => {
```
Comment (nautidpk) [inline auth-connect-frontend/src/components/user-profile.tsx:25]: anyone can go to profile page currently without login

**Review by nautidpk — CHANGES_REQUESTED** [review]

**Comment by nautidpk** [comment]
i have not reviewed whole pr just do quick checking
also rename all your filename as pascal case

File: auth-connect-frontend/pages/signin.tsx (hunk 3 above)
Comment (ayushmi173) [reply auth-connect-frontend/pages/signin.tsx:8]: login is component which is in component folder... Signin is separate page component that inherits llogin

File: auth-connect-frontend/src/components/register.tsx (hunk 7 above)
Comment (ayushmi173) [reply auth-connect-frontend/src/components/register.tsx:70]: I took a reference from paragon Login.tsx.

**Pull request merged**